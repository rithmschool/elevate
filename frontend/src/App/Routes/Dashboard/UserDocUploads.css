.card {
    align-content: center;
    margin: 5rem;
}