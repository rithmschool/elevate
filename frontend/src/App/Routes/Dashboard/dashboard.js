import React from "react";
import { Switch } from "react-router-dom";
import DashboardRoutes from "../dashboardRoutes";
import DashboardManage from "./dashboardManage";
import DashboardAppointments from "./dashboardAppointments";
import DashboardTemplates from "./dashboardTemplates";
import AdminPrivateRoute from "../adminPrivateRoute";
import AdminPanel from "../AdminPanel/adminPanel";

class Dashboard extends React.Component {
  render() {
    return (
      <div>
        <h1>Dashboard</h1>

        <Switch>
          <DashboardRoutes
            exact
            path="/dashboard/manage"
            render={props => <DashboardManage {...props} />}
          />

          <DashboardRoutes
            exact
            path="/dashboard/appointments"
            render={props => <DashboardAppointments {...props} />}
          />

          <DashboardRoutes
            exact
            path="/dashboard/templates"
            render={props => <DashboardTemplates {...props} />}
          />

          <AdminPrivateRoute
            exact={false}
            path="/dashboard/admin"
            render={props => <AdminPanel {...props} />}
          />

          <AdminPrivateRoute
            exact={true}
            path="/dashboard/admin/users/:userId"
            render={props => <AdminPanel {...props} />}
          />
        </Switch>
      </div>
    );
  }
}

export default Dashboard;
