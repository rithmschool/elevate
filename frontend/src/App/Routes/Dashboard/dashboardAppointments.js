import React from "react";

class DashboardAppointments extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <div>
        <h1>Appointments</h1>
      </div>
    );
  }
}

export default DashboardAppointments;
