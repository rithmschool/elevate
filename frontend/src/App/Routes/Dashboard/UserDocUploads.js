import React, { Component } from "react";
import {
  Form,
  FormGroup,
  FormControl,
  Button,
  Alert,
  Card
} from "react-bootstrap";
import ElevateApi from "../../../elevateApi";
import "./UserDocUploads.css"

class UserDocUploads extends Component {
  constructor(props) {
    super(props);
    this.state = {
      files: [],
      uploaded: false
    };

    this.handleUpload = this.handleUpload.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleUpload(e) {
    let files = e.target.files[0];

    if (files) {
      this.setState({ ...this.state, files: [...this.state.files, files] });
    }
  }

  async handleSubmit(e) {
    e.preventDefault();
    this.setState({ ...this.state, uploaded: true });

    const token = localStorage.getItem("token");

    const formData = {
      _token: token,
      title: this.state.files[0].name,
      counterparty: "Alex",
      status: "unreviewed",
      date_reviewed: null
    };

    const result = await ElevateApi.uploadDoc(formData);
  }

  render() {
    return (
      <div className="file-upload-wrapper">
        <Card className="card text-center p-5">
          <Card.Body>
            <Card.Subtitle className="mb-2 text-muted">
              Drop documents here for review
            </Card.Subtitle>
            <div>
              <Form onSubmit={this.handleSubmit}>
                <FormGroup>
                  <FormControl
                    name="docs"
                    type="file"
                    multiple
                    onChange={this.handleUpload}
                  ></FormControl>
                </FormGroup>
                <Button type="submit">Upload a document</Button>
              </Form>
              <div>
                {this.state.uploaded ? (
                  <Alert variant="success">File uploaded</Alert>
                ) : <i className="fas fa-upload fa-2x mt-3"></i>}
              </div>
            </div>
          </Card.Body>
        </Card>
      </div>
    );
  }
}

export default UserDocUploads;
