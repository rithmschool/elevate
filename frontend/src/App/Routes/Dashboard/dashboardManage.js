import React from "react";
import UserDocUploads from "./UserDocUploads";

class DashboardManage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      files: []
    };
    this.handleDrop = this.handleDrop.bind(this);
  }

  handleDrop = files => {
    this.setState({ files });
  };

  render() {
    return (
      <div>
        <UserDocUploads handleDrop={this.handleDrop} />
      </div>
    );
  }
}

export default DashboardManage;
