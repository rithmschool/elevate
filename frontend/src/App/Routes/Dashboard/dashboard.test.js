import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import Dashboard from "./dashboard";

it("renders without crashing", function() {
  shallow(<Dashboard />);
});

it("matches snapshot", function() {
  let wrapper = shallow(<Dashboard />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});
