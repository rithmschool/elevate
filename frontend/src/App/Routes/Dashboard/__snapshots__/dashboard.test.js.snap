// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <h1>
    Dashboard
  </h1>
  <Switch>
    <DashboardRoutes
      exact={true}
      path="/dashboard/manage"
      render={[Function]}
    />
    <DashboardRoutes
      exact={true}
      path="/dashboard/appointments"
      render={[Function]}
    />
    <DashboardRoutes
      exact={true}
      path="/dashboard/templates"
      render={[Function]}
    />
    <AdminPrivateRoute
      exact={false}
      path="/dashboard/admin"
      render={[Function]}
    />
    <AdminPrivateRoute
      exact={true}
      path="/dashboard/admin/users/:userId"
      render={[Function]}
    />
  </Switch>
</div>
`;
